pkg/api/v1/conversion_test.go

```diff
@@ -101,3 +102,54 @@ func TestListOptionsConversion(t *testing.T) {
 		}
 	}
 }
+
+func TestResourceListConversion(t *testing.T) {
+	bigMilliQuantity := resource.NewQuantity(resource.MaxMilliValue, resource.DecimalSI)
+	bigMilliQuantity.Add(resource.MustParse("1m"))
+	bigRoundedQuantity := resource.NewQuantity(resource.MaxMilliValue+1, resource.DecimalSI)
+
+	tests := []struct {
+		input    versioned.ResourceList
+		expected api.ResourceList
+	}{
+		{ // No changes necessary.
+			input: versioned.ResourceList{
+				versioned.ResourceMemory:  resource.MustParse("30M"),
+				versioned.ResourceCPU:     resource.MustParse("100m"),
+				versioned.ResourceStorage: resource.MustParse("1G"),
+			},
+			expected: api.ResourceList{
+				api.ResourceMemory:  resource.MustParse("30M"),
+				api.ResourceCPU:     resource.MustParse("100m"),
+				api.ResourceStorage: resource.MustParse("1G"),
+			},
+		},
+		{ // Nano-core CPU value should be rounded up to millicores.
+			input: versioned.ResourceList{
+				versioned.ResourceCPU: resource.MustParse("3000023n"),
+			},
+			expected: api.ResourceList{
+				api.ResourceCPU: resource.MustParse("4m"),
+			},
+		},
+		{ // Large CPU value should be rounded up to cores.
```

Comment:
Attempting to use millicore precision at a magnitude that would overflow an int64 I think is just asking for trouble, and I don't see the value in it. I didn't bother to document this since I can't imagine anyone ever running into this (good luck scheduling a pod asking for 10-petacores), but I'm happy to add it for completeness' sake.